File: Beecrowd/6-Paradigmas/1932-Bolsa_de_Valores-7/1932.py
s, t = map(int, input().split(' '))
day = list(map(int, input().split(' ')))
ant = current = day[0]
ans = 0
for i in range(1, s):
    if (current > day[i] and current-day[i]>=t) or ant>day[i]:
        ans=max(ans, ans+ current-ant-t)
        current = ant = day[i]
        
    if(day[i]>current):current = day[i]
ans=max(ans, ans + current-ant-t)
print(ans)

File: Beecrowd/6-Paradigmas/1932-Bolsa_de_Valores-7/1932.cs
using System;
using System.Linq;

public class Solution {
    public static void Main(String[] args){
        int s, t, ant, current, ans;
        string[] aux = Console.ReadLine().Split(' ');
        s=int.Parse(aux[0]);
        t=int.Parse(aux[1]);
        int[] days = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
        ans = 0;
        ant = current = days[0];
        for(int i =1;i<s;++i){
            if((current>days[i] && current-days[i]>=t) || ant>days[i]){
                ans = Math.Max(ans, ans+current-ant-t);
                current = ant = days[i];
            }
            if(days[i]>current)current=days[i];
        }
        ans=Math.Max(ans, ans+current-ant-t);
        Console.WriteLine(ans);

    }
}
